### DISCUSSION TITLE
Report of experience - invalid MusicXML file

### DISCUSSION THREAD

**Discussion by grarddeg** [Audiveris/audiveris#662] [Q&A]
After a few days of experience with Audiveris, I find it works reasonably well both on a vectorial pdf (generated by a music engraving software) and on a (good resolution) bitmap pdf (scanned printed score).
I did however encounter a problem with one of the scanned pages, the resulting MusicXML file could not be imported either by MuseScore (invalid MusicXML file) or by musicxml2ly. I enclose the zipped mxl file and the musicxml2ly log mxl file. I can also provide the uncompressed MusicXML file if need be.

[p2.mxl.zip](https://github.com/Audiveris/audiveris/files/12206688/p2.mxl.zip)
musicxml2ly log 
****************
Démarrage musicxml2ly...
musicxml2ly: Reading MusicXML from /Users/grarddeg/Library/AudiverisLtd/audiveris/data/Krommer/p2.mxl ...
musicxml2ly: Input file /Users/grarddeg/Library/AudiverisLtd/audiveris/data/Krommer/p2.mxl is compressed, extracting raw MusicXML data
musicxml2ly: warning: overfull measure? Expected: 269/4, Difference: -1/4
  In: <measure number=67 width=188>

  In: <part id=P1>

  In: <score-partwise version=4.0.2>

musicxml2ly: warning: overfull measure? Expected: 269/4, Difference: -1/4
  In: <measure number=67>

  In: <part id=P2>

  In: <score-partwise version=4.0.2>

musicxml2ly: warning: overfull measure? Expected: 269/4, Difference: -1/4
  In: <measure number=67>

  In: <part id=P3>

  In: <score-partwise version=4.0.2>

musicxml2ly: warning: overfull measure? Expected: 269/4, Difference: -1/4
  In: <measure number=67>

  In: <part id=P4>

  In: <score-partwise version=4.0.2>

musicxml2ly: warning: overfull measure? Expected: 269/4, Difference: -1/4
  In: <measure number=67>

  In: <part id=P5>

  In: <score-partwise version=4.0.2>

musicxml2ly: warning: overfull measure? Expected: 269/4, Difference: -1/4
  In: <measure number=67>

  In: <part id=P6>

  In: <score-partwise version=4.0.2>

musicxml2ly: warning: overfull measure? Expected: 269/4, Difference: -1/4
  In: <measure number=67>

  In: <part id=P7>

  In: <score-partwise version=4.0.2>

musicxml2ly: warning: overfull measure? Expected: 269/4, Difference: -1/4
  In: <measure number=67>

  In: <part id=P8>

  In: <score-partwise version=4.0.2>

musicxml2ly: warning: overfull measure? Expected: 269/4, Difference: -1/4
  In: <measure number=67>

  In: <part id=P9>

  In: <score-partwise version=4.0.2>

musicxml2ly: warning: overfull measure? Expected: 269/4, Difference: -1/4
  In: <measure number=67>

  In: <part id=P10>

  In: <score-partwise version=4.0.2>

musicxml2ly: warning: overfull measure? Expected: 269/4, Difference: -1/4
  In: <measure number=67>

  In: <part id=P11>

  In: <score-partwise version=4.0.2>

musicxml2ly: Converting to LilyPond expressions...
musicxml2ly: warning: Negative skip -1/4 (from position 269/4 to 67)
musicxml2ly: warning: Negative skip -1/4 (from position 269/4 to 67)
musicxml2ly: warning: Negative skip -1/4 (from position 135/2 to 269/4)
musicxml2ly: Converting to LilyPond expressions...
Traceback (most recent call last):
  File "/opt/local/bin/musicxml2ly", line 3482, in <module>
    main()
  File "/opt/local/bin/musicxml2ly", line 3475, in main
    voices = convert(filename, options)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/local/bin/musicxml2ly", line 3336, in convert
    (voices, staff_info) = get_all_voices(parts)
                           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/local/bin/musicxml2ly", line 2923, in get_all_voices
    voice = musicxml_voice_to_lily_voice(v)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/local/bin/musicxml2ly", line 2548, in musicxml_voice_to_lily_voice
    main_event = n.to_lily_object(
                 ^^^^^^^^^^^^^^^^^
  File "/opt/local/share/lilypond/2.24.1/python/musicxml.py", line 892, in to_lily_object
    event.duration = self.initialize_duration()
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/local/share/lilypond/2.24.1/python/musicxml.py", line 841, in initialize_duration
    % (mxl_note.start, mxl_note._duration))
       ^^^^^^^^
NameError: name 'mxl_note' is not defined. Did you mean: 'Xml_node'?
Arrêté avec le code de retour 1.

**Comment by hbitteur** [comment]
When loading your .mxl file, MuseScore reports this error:
> Fatal error: line 1779 column 18 Content of element duration does not match its type definition: Double content does not match the minExclusive facet..

![image](https://github.com/Audiveris/audiveris/assets/22053634/05e0aedc-edbe-4734-b20b-21edc469b03a)

Indeed a 0-duration rest is a very short rest! :-)

Could you post the input file (just the page which contains this measure 63), so that I can investigate further?

**Reply by grarddeg** [reply]
There you go. On measure 63, it didn't catch the 1-note d apparently.

[Krommerp2.pdf](https://github.com/Audiveris/audiveris/files/12206849/Krommerp2.pdf)

**Comment by hbitteur** [comment]
Yes, the double note was not recognized, though the related template seems to fit rather well. To be further checked.

Anyway, an empty measure should not export a 0-duration rest, this is a real bug. To be fixed.

**Comment by grarddeg** [comment]
As a follow-on, setting the duration to 48 wherever it is set to zero works, at least with musicxml2ly, the file gets imported into LilyPond. But I couldn't import it into MuseScore, maybe because both the issue below.
There is strange behaviour though, that I experienced also with the other scanned page : it creates as many voices (or parts) as there are staves in the original score and scatters the music between the voices/parts. This requires to copy each bit into a single voice and remove all others when it's done, which is a little tedious.
